tests/test_loaders.py

```diff
@@ -217,151 +191,174 @@ def test_download(mocker, httpserver):
 
 # This is magically skipped by the the remote fixture `skip_local` in conftest.py
 # when tests are run with the --local flag
-def test_validate(skip_local):
+def test_validate(skip_local, httpserver):
     for dataset_name in DATASETS:
+        data_home = os.path.join("tests/resources/mir_datasets", dataset_name)
+
+        module = importlib.import_module("mirdata.datasets.{}".format(dataset_name))
         if dataset_name not in REMOTE_DATASETS:
-            data_home = os.path.join("tests/resources/mir_datasets", dataset_name)
-            dataset = mirdata.Dataset(dataset_name, data_home=data_home)
-            try:
-                dataset.validate()
-            except:
-                assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
+            dataset = module.Dataset(data_home)
+            dataset_default = module.Dataset(data_home=None)
+        else:
+            remote_index = create_remote_index(httpserver, dataset_name)
+            dataset = module.Dataset(data_home, index=remote_index)
+            dataset_default = module.Dataset(data_home=None, index=remote_index)
 
-            try:
-                dataset.validate(verbose=False)
-            except:
-                assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
+        try:
+            dataset.validate()
+        except:
+            assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
 
-            dataset_default = mirdata.Dataset(dataset_name, data_home=None)
-            try:
-                dataset_default.validate(verbose=False)
-            except:
-                assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
+        try:
+            dataset.validate(verbose=False)
+        except:
+            assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
+
+        try:
+            dataset_default.validate(verbose=False)
+        except:
+            assert False, "{}: {}".format(dataset_name, sys.exc_info()[0])
 
 
 def test_load_and_trackids(httpserver):
     for dataset_name in DATASETS:
         data_home = os.path.join("tests/resources/mir_datasets", dataset_name)
+        module = importlib.import_module("mirdata.datasets.{}".format(dataset_name))
         if dataset_name not in REMOTE_DATASETS:
-            dataset = mirdata.Dataset(dataset_name, data_home=data_home)
-            dataset_default = mirdata.Dataset(dataset_name, data_home=None)
+            dataset = module.Dataset(data_home)
+            dataset_default = module.Dataset()
+        else:
+            continue
+            # TODO - fix the dataset object to work with remote index
+            # remote_index = create_remote_index(httpserver, dataset_name)
+            # dataset = module.Dataset(data_home, index=remote_index)
+            # dataset_default = module.Dataset(index=remote_index)
```

Comment:
This is done already right?

Reply:
not anymore...I re-broke it 🙃 the tests dont work and they need to be updated. It works for the end-user but it's no longer tested